cmd/lakectl/cmd/root.go

```diff
@@ -245,9 +248,11 @@ func getSyncFlags(cmd *cobra.Command, client *apigen.ClientWithResponses) local.
 	if parallelism < 1 {
 		DieFmt("Invalid value for parallelism (%d), minimum is 1.\n", parallelism)
 	}
+	setParallelism := cmd.Flags().Changed(parallelismFlagName)
 
 	presignMode := getPresignMode(cmd, client)
 	return local.SyncFlags{
+		SetParallelism:   setParallelism,
```

Comment:
You don't really need SetParallelism anymore.
Note that you have cfg in your context here, so basically you can make the decision (what value to use for the parallelism) here.